Source/GoogleMapsCarbonFootprint.js

```diff
@@ -233,10 +270,51 @@ function convertDistance(distanceStr) {
  * Return:
  *   - the carbon footprint, in grams.
  */
-function computeFootprint(distance) {
-  var footprint = distance * carbonEmission;
-  console.log('Carbon Footprint for this route is: ' + footprint + ' grams');
-  return footprint;
+
+ /*This function converts the time string to integer Time values */
+function convertTime(timestr) {
+
+    var str_new = timestr.split(" ");
+    console.log(str_new);
+
+    if (str_new[1] == "min") {
+        var time = parseInt(str_new[0]) / 60;
+    } else if (str_new[1] == "h" && str_new[3] != "min") {
+        var time = parseInt(str_new[0]);
+
+    } else if (str_new[1] == "h" && str_new[3] == "min") {
+        var time = parseInt(str_new[0]) + parseInt(str_new[2]) / 60;
+    } else {
+        var time = -1;
+    }
+    console.log("fimnal time is" + time);
+    return time;
+}
+
+
+function computeFootprint(distance, time) {
+
+    /*CALCULATE AVERAGE SPEED AS PER GOOGLE MAPS*/
+    if (time == -1) { averageSpeed = 30; } else {
+        var averageSpeed = distance / time;
+    }
+    console.log("average" + averageSpeed);
+
+    /*User given carbon emmission are taken on the basis on the average speed*/
```

Comment:
This estimation is not precise. Note that your function will compute the same footprint for average speed is 60 and 89. In other words, you are assuming that the emission rate is constant between 10 and 30, between 30 and 60 and between 60 and 90. That is not a reasonable assumption.

It would be better to do linear interpolation. For example, if you know that the carbon emission rate at 60km/h is 200g/km and at 90km/h it is 300g/km, then the carbon emission rate at an average speed x between 60km/h and 90km/h would be:

e = 200 + (300 - 200) * (x - 60) / (90 - 60)